src/Simplex/Messaging/Agent/Store/SQLite/Util.hs

```diff
@@ -1,382 +1,403 @@
-{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DuplicateRecordFields #-}
-{-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE FlexibleInstances #-}
-{-# LANGUAGE GADTs #-}
-{-# LANGUAGE InstanceSigs #-}
 {-# LANGUAGE LambdaCase #-}
-{-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE ScopedTypeVariables #-}
-{-# LANGUAGE TypeApplications #-}
-{-# LANGUAGE UndecidableInstances #-}
 {-# OPTIONS_GHC -fno-warn-orphans #-}
 
-module Simplex.Messaging.Agent.Store.SQLite.Util where
-
-import Control.Monad
-import Control.Monad.Except
-import Control.Monad.IO.Unlift
-import Data.Int (Int64)
+module Simplex.Messaging.Agent.Store.SQLite.Util
+  ( createRcvQueueAndConn,
+    createSndQueueAndConn,
+    retrieveConnQueues,
+    retrieveRcvQueue,
+    deleteConnCascade,
+    updateRcvConnWithSndQueue,
+    updateSndConnWithRcvQueue,
+    updateRcvQueueStatus,
+    updateSndQueueStatus,
+    insertRcvMsg,
+    insertSndMsg,
+  )
+where
+
+import Control.Monad.Except (MonadIO (liftIO))
+import Data.Maybe (fromMaybe)
 import qualified Data.Text as T
-import Data.Time
-import Database.SQLite.Simple hiding (Connection)
-import qualified Database.SQLite.Simple as DB
+import Data.Time (getCurrentTime)
+import Database.SQLite.Simple as DB
 import Database.SQLite.Simple.FromField
 import Database.SQLite.Simple.Internal (Field (..))
-import Database.SQLite.Simple.Ok
+import Database.SQLite.Simple.Ok (Ok (Ok))
 import Database.SQLite.Simple.QQ (sql)
-import Database.SQLite.Simple.ToField
-import Network.Socket
+import Database.SQLite.Simple.ToField (ToField (..))
+import Network.Socket (HostName, ServiceName)
 import Simplex.Messaging.Agent.Store
 import Simplex.Messaging.Agent.Store.Types
 import Simplex.Messaging.Agent.Transmission
-import Simplex.Messaging.Protocol as SMP
-import Simplex.Messaging.Util
-import Text.Read
+import Simplex.Messaging.Protocol as SMP (RecipientId)
+import Text.Read (readMaybe)
 import qualified UnliftIO.Exception as E
-import UnliftIO.STM
 
-data SQLiteStore = SQLiteStore
-  { dbFilename :: String,
-    conn :: DB.Connection,
-    serversLock :: TMVar (),
-    rcvQueuesLock :: TMVar (),
-    sndQueuesLock :: TMVar (),
-    connectionsLock :: TMVar (),
-    messagesLock :: TMVar ()
-  }
+-- ? replace with ToField? - it's easy to forget to use this
+serializePort_ :: Maybe ServiceName -> ServiceName
+serializePort_ = fromMaybe "_"
 
-type QueueRowId = Int64
+deserializePort_ :: ServiceName -> Maybe ServiceName
+deserializePort_ port
+  | port == "_" = Nothing
+  | otherwise = Just port
```

Comment:
In such cases it's better to use pattern matches than guards:

```
deserializePort_ "_" = Nothing
deserializePort_ port = Just port
```

or

```
deserializePort_ = \case
  "_" -> Nothing
  port -> Just port
```

guards are usually for conditions when pattern matches don't work